kmmsdk/src/commonMain/kotlin/com/lokate/kmmsdk/LokateSDK.kt

```diff
@@ -1,103 +1,227 @@
+                }
+            }
+        }
+    }
+
+    private fun checkAppToken() {
+        lokateScope.launch {
+            authenticationRepository.getAppToken().let {
+                if (it is RepositoryResult.Success && it.body.isNotEmpty()) {
+                    appTokenSet = true
+                }
+            }
+        }
+    }
+
+    init {
+        log.d { "LokateSDK initializing" }
+        // check if app token is set
+        checkAppToken()
+
+        if (!appTokenSet) {
+            log.e { "App token not set" }
+            // use default token
+            setAppToken("1")
+        }
+    }
+
     fun startScanning() {
         beaconScanner.setRegions(DEFAULT_BEACONS)
+        //
+        fetchActiveBeacons(EMPTY_STRING)
+        //
         beaconScanner.startScanning()
 
-        val beaconScannerFlow = beaconScanner.scanResultFlow()
-        val newComer = beaconScannerFlow.filter { it !in activeBeacons }
-        val alreadyIn =
-            beaconScannerFlow.filter { it in activeBeacons }.transform {
-                emit(
-                    activeBeacons.find {
-                        it.beaconUUID == it.beaconUUID && it.major == it.major && it.minor == it.minor
-                    }?.copy(lastSeen = getTimeMillis()),
-                )
-            }
-        // in the last 3 seconds, if the beacon is not in the list, it is considered gone
-        val gone =
-            flow {
+        scanResultHandler(
+            beaconScannerFlow =
+                beaconScanner.scanResultFlow().transform { scan ->
+                    val beacon =
+                        branchBeacons.firstOrNull {
+                            it.id.lowercase() == scan.beaconUUID.lowercase() &&
+                                it.major == scan.major.toString() &&
+                                it.minor == scan.minor.toString()
+                        }
+                    val comparison = (beacon == null || beacon.range.ordinal >= scan.proximity.ordinal)
+                    if (scan.accuracy != null && scan.accuracy > -1.0 && comparison) {
+                        emit(scan)
+                    }
+                },
+        )
+        checkGone()
+        flowEmitter()
+        // beaconScanner.start()
+        // results = beaconScanner.observeResults()
+    }
+
+    private fun checkGone() {
+        lokateScope.launch {
+            while (true) {
+                delay(GONE_CHECK_INTERVAL)
+                log.d { "Checking for gone beacons" }
                 activeBeacons.forEach {
-                    if (it.lastSeen < getTimeMillis() - DEFAULT_TIMEOUT_BEFORE_GONE) {
-                        emit(it)
+                    if (it.seen < getTimeMillis() - Defaults.DEFAULT_TIMEOUT_BEFORE_GONE) {
+                        gone.emit(it)
+                        activeBeacons.remove(it)
                     }
                 }
             }
+        }
+    }
 
+    private fun flowEmitter() {
```

Comment:
name seems misleading, this function does not emit values. it can be flowCollector()